File: lex-c/app-dev/src/AppStateManager.cpp
#include "AppStateManager.h"
#include "StringVector.h"

AppStateManager::AppStateManager() {
    this->appState.stepNo = 0;
    this->appState.lineNo = 0;
    this->appState.currentPosInLine = 0;
};
AppState AppStateManager::getAppState() {
    return appState;
};
StringVector AppStateManager::getLimiters() {
    return this->appState.limiters;
};
StringVector AppStateManager::getSpaces() {
    return this->appState.spaces;
};
AppStateManager *AppStateManager::setStepNo(unsigned int stepNo) {
    this->appState.stepNo = stepNo;
    return this;
};
AppStateManager *AppStateManager::setLineNo(unsigned int lineNo) {
    this->appState.lineNo = lineNo;
    return this;
};
AppStateManager *AppStateManager::setCurrentPosInLine(unsigned int currentPosInLine) {
    this->appState.currentPosInLine = currentPosInLine;
    return this;
};
AppStateManager *AppStateManager::setInputString(std::string inputString) {
    this->appState.inputString = inputString;
    return this;
}
AppStateManager *AppStateManager::setSpaces(StringVector spaces) {
    this->appState.spaces = spaces;
    return this;
};
AppStateManager *AppStateManager::setLimiters(StringVector limiters) {
    this->appState.limiters = limiters;
    return this;
};
AppStateManager *AppStateManager::setIds(StringVector ids) {
    this->appState.ids = ids;
    return this;
};
AppStateManager *AppStateManager::setStrings(StringVector strings) {
    this->appState.strings = strings;
    return this;
};
AppStateManager *AppStateManager::setCompiled(CompiledLineVector compiled) {
    this->appState.compiled = compiled;
    return this;
}
AppStateManager *AppStateManager::setText(CanonicTextItemVector text) {
    this->appState.text = text;
    return this;
}
unsigned int AppStateManager::getStepNo() {
    return this->appState.stepNo;
};
std::string AppStateManager::getInputString() {
    return this->appState.inputString;
}
CompiledLineVector AppStateManager::getCompiled() {
    return this->appState.compiled;
}
CanonicTextItemVector AppStateManager::getText() {
    return this->appState.text;
}

File: lex-feconf-draft/app-dev/src/AppStateManager.ts
import { store } from '@src/store/store';
import { app } from '@src/store/appReducer';
import {
    CanonicTextItem,
    CompiledLine,
    defaultSyntaxAnalyzeState,
    SyntaxAnalyzeState
} from './app.types';

export class AppStateManager {
    private stepNo: number = 0;
    private spaces: string[] = [];
    private limiters: string[] = [];
    private ids: string[] = [];
    private strings: string[] = [];
    private compiled: CompiledLine[];
    private inputString: string = '';
    private lineNo: number = 0;
    private currentPosInLine: number = 0;
    private text: CanonicTextItem[] = [];
    private program: SyntaxAnalyzeState = { ...defaultSyntaxAnalyzeState };
    private consoleText: string = '';

    getStepNo = () => this.stepNo;
    getSpaces = () => this.spaces;
    getLimiters = () => this.limiters;
    getIds = () => this.ids;
    getStrings = () => this.strings;
    getCompiled = () => this.compiled;
    getInputString = () => this.inputString;
    getLineNo = () => this.lineNo;
    getCurrentPosInLine = () => this.currentPosInLine;
    getText = () => this.text;
    getProgram = () => this.program;
    getConsoleText = () => this.consoleText;

    setStepNo = (stepNo: number) => (this.stepNo = stepNo);
    setSpaces = (spaces: string[]) => (this.spaces = spaces);
    setLimiters = (limiters: string[]) => (this.limiters = limiters);
    setIds = (ids: string[] = []) => (this.ids = ids);
    setStrings = (strings: string[] = []) => (this.strings = strings);
    setCompiled = (compiled: CompiledLine[]) => (this.compiled = compiled);
    setInputString = (inputString: string) => (this.inputString = inputString);
    setLineNo = (lineNo: number) => (this.lineNo = lineNo);
    setCurrentPosInLine = (currentPosInLine: number) => (this.currentPosInLine = currentPosInLine);
    setText = (text: CanonicTextItem[]) => (this.text = text);
    setProgram = (program: SyntaxAnalyzeState) => (this.program = program);
    setConsoleText = (consoleText: string) => (this.consoleText = consoleText);

    getAppState = () => {
        return {
            stepNo: this.getStepNo(),
            spaces: this.getSpaces(),
            limiters: this.getLimiters(),
            ids: this.getIds(),
            compiled: this.getCompiled(),
            inputString: this.getInputString(),
            lineNo: this.getLineNo(),
            currentPosInLine: this.getCurrentPosInLine(),
            text: this.getText(),
            program: this.getProgram()
        };
    };

    mirrorState = () => {
        store.dispatch(app.setSimState(this.getAppState()));
    };
}
